include/vcpkg/base/stringview.h

```diff
@@ -86,3 +87,47 @@ namespace vcpkg
 
 VCPKG_FORMAT_AS(vcpkg::ZStringView, vcpkg::StringView);
 VCPKG_FORMAT_AS(vcpkg::StringLiteral, vcpkg::StringView);
+
+template<std::size_t N, std::size_t... Is>
+constexpr std::array<char, N - 1> to_array(const char (&a)[N], std::index_sequence<Is...>)
+{
+    return {{a[Is]...}};
+}
+
+template<std::size_t N>
+constexpr std::array<char, N - 1> to_array(const char (&a)[N])
+{
+    return to_array(a, std::make_index_sequence<N - 1>());
+}
```

Comment:
I don't understand what this is trying to do. If I'm reading it correctly, it always gives you a std::array substantially the same as the input, so you could have said `std::array{a}` for the same effect. Or alternately just left it as an array. Did you mean to do null termination or something in here or...?